### PR TITLE
Switch from semaphore to CircleCI #170582273

### PR DESCRIPTION
**Pull request by software-project**
https://www.pivotaltracker.com/story/show/170582273

### CODE REVIEW

File: spec/javascript/utils/SetupBrowserAndPage.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const IgnoreImageAndCSSLoad = async (page) => {
 }
 
 const SetupBrowser = async () => {
-  let browser = await puppeteer.launch({ headless })
```
Comment (akegan) [inline spec/javascript/utils/SetupBrowserAndPage.js:18]: this looks not right. headless and removecss should be separate things and we should still be indicating whether or not it's headless?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            wget localhost:3000
+            cat index.html
+      - run: yarn e2e
+  coverage:
+    executor: ruby_node
+    steps:
+      - checkout
+      - *attach_workspace
+      # - run: ./tmp/cc-test-reporter format-coverage -t lcov -o tmp/codeclimate.frontend.json coverage/frontend/.resultset.json
+      # - run: ./tmp/cc-test-reporter format-coverage -t simplecov -o tmp/codeclimate.backend.json coverage/backend/.resultset.json
```
Comment (akegan) [inline .circleci/config.yml]:
```suggestion
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    steps:
+      - checkout
+      - *attach_workspace
+      - *restore_cache
+      - puppeteer/install
+      - run:
+          name: install yarn dependencies
+          command: yarn install --check-files
+      - run: bundle --path vendor/bundle
+      - run: bundle exec rake assets:precompile
```
Comment (akegan) [inline .circleci/config.yml:110]: Now that we are persisting state correctly across jobs, we probably don't need to re-install packages, right?

Comment (software-project) [reply .circleci/config.yml:110]: I'll give it a try

File: spec/javascript/utils/SetupBrowserAndPage.js (hunk 1 above)
Comment (akegan) [reply spec/javascript/utils/SetupBrowserAndPage.js:18]: this is still an issue. we need to be able to run non-headless locally. I'll fix it

**Pull request merged**